cmake/Modules/FindSYCLToolkit.cmake

```diff
@@ -1,54 +1,97 @@
-# This will define the following variables:
-# SYCL_FOUND               : True if the system has the SYCL library.
-# SYCL_INCLUDE_DIR         : Include directories needed to use SYCL.
-# SYCL_LIBRARY_DIR         ：The path to the SYCL library.
-# SYCL_LIBRARY             : SYCL library fullname.
+#[=======================================================================[.rst:
+SYCLConfig
+-------
 
-include(FindPackageHandleStandardArgs)
+Library to verify SYCL compatability of CMAKE_CXX_COMPILER
+and passes relevant compiler flags.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``SYCLTOOLKIT_FOUND``
+  True if the system has the SYCL library.
+``SYCL_COMPILER``
+  SYCL compiler executable.
+``SYCL_INCLUDE_DIR``
+  Include directories needed to use SYCL.
+``SYCL_LIBRARY_DIR``
+  Libaray directories needed to use SYCL.
+``SYCL_FLAGS``
+  SYCL specific flags for the compiler.
+``SYCL_LANGUAGE_VERSION``
+  The SYCL language spec version by Compiler.
+
+``SYCL::SYCL_CXX``
+  Interface target for using SYCL compiler.  The following properties are
+  defined for the target: ``INTERFACE_COMPILE_OPTIONS``,
+  ``INTERFACE_LINK_OPTIONS``, ``INTERFACE_INCLUDE_DIRECTORIES``, and
+  ``INTERFACE_LINK_DIRECTORIES``
+
+#]=======================================================================]
+
+set(SYCLTOOLKIT_FOUND False)
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
 
 set(SYCL_ROOT "")
 if(DEFINED ENV{SYCL_ROOT})
   set(SYCL_ROOT $ENV{SYCL_ROOT})
 elseif(DEFINED ENV{CMPLR_ROOT})
   set(SYCL_ROOT $ENV{CMPLR_ROOT})
 endif()
+if(NOT SYCL_ROOT)
+  execute_process(
```

Comment:
cmake find_program can help? https://cmake.org/cmake/help/latest/command/find_program.html

Reply:
Logic of `find_program` is to find a program basing on a known path. And logic here is to figure out a path from a known program execution.